py3status/modules/weekcal.py

```diff
@@ -0,0 +1,156 @@
+    {'color': '#00FF00', 'full_text': '26 '},
+    {'color': '#FFFFFF', 'full_text': '27 28 '},
+    {'color': '#FFFF00', 'full_text': '29 30'},
+    {'color': '#FFFFFF', 'full_text': 'Wed Apr ', 'separator': True},
+    {'color': '#00FF00', 'full_text': '26 '},
+]
+
+custom-2
+[
+    {'color': '#999999', 'full_text': '23 '},
+    {'color': '#FFFFFF', 'full_text': '24 25 '},
+    {'color': '#00FF00', 'full_text': 'Wed Apr 26 '},
+    {'color': '#FFFFFF', 'full_text': '27 28 '},
+    {'color': '#999999', 'full_text': '29'},
+    {'color': '#00FF00', 'full_text': 'Wed Apr 26', 'separator': True},
+]
+
+custom-3
+[
+    {'color': '#FFFFFF', 'full_text': 'Wed Apr '},
+    {'color': '#999999', 'full_text': '23 24 25 '},
+    {'color': '#FFFFFF', 'full_text': '26 '},
+    {'color': '#999999', 'full_text': '27 28 29'},
+    {'color': '#FFFFFF', 'full_text': 'Wed Apr 26', 'separator': True},
+]
+"""
+
+from calendar import Calendar
+from datetime import datetime, date, timedelta
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    button_toggle = 1
+    first_weekday = 6
+    format = '%a %b {weekcal}'
+    format_separator = ' '
+    format_today = '\?color=good %d'
+    format_weekday = '%d'
+    format_weekend = '%d'
+    is_toggled = False
+
+    def post_config_hook(self):
+        self.calendar = Calendar(self.first_weekday)
+
+    def weekcal(self):
+        today = date.today()
+        data = []
+        nextweek = False
+
+        weekdays = self.calendar.iterweekdays()
+        for weekday_number in weekdays:
+            if weekday_number == 0 and self.first_weekday != 0:
+                nextweek = True
+            if nextweek and today.weekday() >= self.first_weekday:
+                weekday_number += 7
+            elif not nextweek and today.weekday() < self.first_weekday:
+                weekday_number -= 7
+
+            weekday_offset = today.weekday() - weekday_number
+            weekday_delta = timedelta(days=weekday_offset)
+            weekday = today - weekday_delta
+            _format = None
+
+            if weekday == today:
+                _format = self.format_today
+            elif not self.is_toggled:
+                if weekday.weekday() <= 4:
+                    _format = self.format_weekday
+                else:
+                    _format = self.format_weekend
+            if _format:
+                data.append(self.py3.safe_format(weekday.strftime(_format)))
+
+        weekcal = self.py3.composite_join(self.format_separator, data)
+        full_text = self.py3.safe_format(
+            today.strftime(self.format), {'weekcal': weekcal})
+
+        # seconds until midnight
+        now = datetime.now()
+        tomorrow = now + timedelta(1)
+        midnight = datetime(year=tomorrow.year, month=tomorrow.month,
+                            day=tomorrow.day, hour=0, minute=0, second=0)
+        cache_timeout = (midnight - now).seconds
```

Comment:
This although seeming logical is overcomplicating things and also inaccurate.  It will fail for example when the clocks go back/forward.  Would it not just be easier to sync to the next hour `self.py3.time_in(sync_to=3600)`